Libraries/LibWeb/Painting/Paintable.cpp

```diff
@@ -165,13 +165,10 @@ void Paintable::set_needs_display(InvalidateDisplayList should_invalidate_displa
     if (should_invalidate_display_list == InvalidateDisplayList::Yes)
         document.invalidate_display_list();
 
-    auto* containing_block = this->containing_block();
-    if (!containing_block)
+    auto* parent = as_if<Painting::PaintableWithLines>(this->parent());
+    if (!parent)
```

Comment:
Commit message: please don't [use weasel-words such as "fix"](https://github.com/LadybirdBrowser/ladybird/blob/master/CONTRIBUTING.md#code-submission-policy), but rather describe what you changed and why.